base/abstractarray.jl

```diff
@@ -37,7 +37,7 @@ Base.OneTo(6)
 """
 function indices(A::AbstractArray{T,N}, d) where {T,N}
     @_inline_meta
-    d <= N ? indices(A)[d] : OneTo(1)
+    d <= N ? indices(A)[d] : one(indicestype(typeof(A),d))
```

Comment:
That would indeed simplify things. It's what I'm doing right now as a workaround. So far I noticed this breaks `display(A)`, but I'm sure that's fixable.